examples/example_DictionaryMatchOp.py

```diff
@@ -0,0 +1,154 @@
+
+# %% [markdown]
+# As a loss function for the optimizer, we calculate the mean-squared error between the image data $x$ and our signal
+# model $q$.
+# %%
+mse = MSEDataDiscrepancy(idata_multi_ti.data.abs())
+
+# %% [markdown]
+# Now we can simply combine the two into a functional to solve
+#
+# $ \min_{M_0, T1} || |q(M_0, T1, TI)| - x||_2^2$
+# %%
+functional = mse @ model
+
+# %% [markdown]
+# ### Starting values for the fit
+# We are trying to minimize a non-linear function $q$. There is no guarantee that we reach the global minimum, but we
+# can end up in a local minimum.
+#
+# To increase our chances of reaching the global minimum, we can ensure that our starting
+# values are already close to the global minimum. We need a good starting point for each pixel.
+#
+# One option to get a good starting point is to calculate the signal curves for a range of T1 values and then check
+# for each pixel which of these signal curves fits best. This is similar to what is done for MR Fingerprinting. So we
+# are going to:
+# - define a list of realistic T1 values (we call this a dictionary of T1 values)
+# - calculate the signal curves corresponding to each of these T1 values
+# - compare the signal curves to the signals of each voxel (we use the maximum of the dot-product as a metric of how
+# well the signals fit to each other)
+
+# %%
+# Define 100 T1 values between 100 and 3000 ms
+t1_dictionary = torch.linspace(100, 3000, 100)
+print(type(t1_dictionary))
+
+# Calculate the signal corresponding to each of these T1 values. We set M0 to 1, but this is arbitrary because M0 is
+# just a scaling factor and we are going to normalize the signal curves.
+(signal_dictionary,) = model(torch.ones(1), t1_dictionary)
+signal_dictionary = signal_dictionary.to(dtype=torch.complex64)
+vector_norm = torch.linalg.vector_norm(signal_dictionary, dim=0)
+signal_dictionary /= vector_norm
+
+# Calculate the dot-product and select for each voxel the T1 values that correspond to the maximum of the dot-product
+n_y, n_x = idata_multi_ti.data.shape[-2:]
+dot_product = torch.mm(rearrange(idata_multi_ti.data, 'other 1 z y x->(z y x) other'), signal_dictionary)
+# print(signal_dictionary)
+idx_best_match = torch.argmax(torch.abs(dot_product), dim=1)
+# print(torch.abs(dot_product))
+t1_start = rearrange(t1_dictionary[idx_best_match], '(y x)->1 1 y x', y=n_y, x=n_x)
+
+
+Tin = TypeVarTuple('Tin')
+
+
+class DictionaryMatchOp(Operator[torch.Tensor, tuple[*Tin]]):
+    def __init__(self, generating_function: Callable[[Unpack[Tin]], tuple[torch.Tensor,]]):
+        super().__init__()
+        self._f = generating_function
+        self.x: list[torch.Tensor] = []
+        self.y = torch.tensor([])
+
+    def append(self, *x: Unpack[Tin]) -> Self:
+        (newy,) = self._f(*x)
+        newy = newy.to(dtype=torch.complex64)
+        newy = newy / torch.linalg.norm(newy, dim=0, keepdim=True)
+        newy = newy.flatten(start_dim=1)
+        newx = x[1]
```

Comment:
Save all the x values, not only the second parameter

So self.x is a list of all the parameters:
Here: [M0,T1]